assets/js/components/dashboard-sharing/ModuleRecoveryAlert/index.js

```diff
@@ -72,198 +61,48 @@ export default function ModuleRecoveryAlert() {
 			.map( ( { slug } ) => slug );
 	} );
 
-	const recoveryErrors = useSelect( ( select ) => {
-		if ( ! recoverableModules ) {
-			return undefined;
-		}
-
-		const recoveredModules = select( CORE_MODULES ).getRecoveredModules();
-
-		if ( ! recoveredModules ) {
-			return {};
-		}
-
-		const modules = Object.keys( recoverableModules );
-
-		const getRecoveryError = ( module ) =>
-			recoveredModules?.error?.[ module ];
-
-		return modules
-			.filter( ( module ) => !! getRecoveryError( module ) )
-			.reduce(
-				( acc, module ) => ( {
-					...acc,
-					[ module ]: {
-						name: recoverableModules[ module ].name,
-						...getRecoveryError( module ),
-					},
-				} ),
-				{}
-			);
-	} );
-
-	const { recoverModules, clearRecoveredModules } =
-		useDispatch( CORE_MODULES );
-
-	const isLoading =
-		userAccessibleModules === undefined || checkboxes === null;
-
-	const updateCheckboxes = useCallback(
-		( slug ) =>
-			setCheckboxes( ( currentCheckboxes ) => ( {
-				...currentCheckboxes,
-				[ slug ]: ! currentCheckboxes[ slug ],
-			} ) ),
-		[]
+	// The alert renders conditional copy and actions based on:
+	// 1. If there is one or more than one module to recover.
+	// 2. If the user has access to perform the recovery.
+	const hasMultipleRecoverableModules = useMemo(
+		() => Object.keys( recoverableModules || {} ).length > 1,
+		[ recoverableModules ]
+	);
+	const hasUserRecoverableModules = useMemo(
+		() => !! Object.keys( userAccessibleModules || {} ).length,
+		[ userAccessibleModules ]
```

Comment:
- +1 to "user recoverable modules" as this is what I would have used as well. Please also update `userAccessibleModules` above as well since name is more accurate and applies there as well.
- I'm not sure `useMemo` is providing a benefit here when used with `[ userAccessibleModules ]` as the dependencies since it returns a new array every time (so it's likely never returning the memoized value). I think it should be okay to skip using this hook here anyways since the computation is probably less expensive than the overhead of the hook.